plugins/ndenv/ndenv.plugin.zsh

```diff
@@ -0,0 +1,42 @@
+_homebrew-installed() {
+  type brew &> /dev/null
+}
+
+FOUND_NDENV=0
+ndenvdirs=("$HOME/.ndenv" "/usr/local/ndenv" "/opt/ndenv" "/usr/local/opt/ndenv")
+
+if _homebrew-installed && ndenv_homebrew_path=$(brew --prefix ndenv 2>/dev/null); then
+    ndenvdirs=($ndenv_homebrew_path "${ndenvdirs[@]}")
+    unset ndenv_homebrew_path
+fi
+
+for ndenvdir in "${ndenvdirs[@]}" ; do
+  if [ -d $ndenvdir/bin -a $FOUND_NDENV -eq 0 ] ; then
+    FOUND_NDENV=1
+
+    if [[ $NDENV_ROOT = '' ]]; then
```

Comment:
If `$NDENV_ROOT` is already set by the caller, should that just be used directly, instead of searching for the installation location? Here, you get a situation where `$NDENV_ROOT` and `$PATH` may not match, if the caller supplied `$NDENV_ROOT` but it isn't the first extant location in your search order.